### PR TITLE
refactor: use config(Cache::class) in CodeIgniter

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
- refactor CodeIgniter
- improve test

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPDoc blocks, only if necessary or adds value
- [] Unit testing, with >80% coverage
- [] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Comment by mostafakhudair** [comment]
`CodeIgniter\CodeIgniter::display404errors()` should be affected too

**Comment by kenjis** [comment]
Updated `:display404errors()`.

**Comment by kenjis** [comment]
Please review.

File: tests/system/CodeIgniterTest.php [hunk 1]
```diff
@@ -792,9 +793,15 @@ public function testPageCacheWithCacheQueryString(
...
             // with value from the data provider
+            ob_start();
             $this->codeigniter->run();
-            // Cache the page using our own $cacheConfig confugration
-            $this->codeigniter->cachePage($cacheConfig);
+            $output = ob_get_clean();
+
+            $this->assertSame($string, $output);
+
+            if (count(ob_list_handlers()) > 1) {
```
Comment (MGatner) [inline tests/system/CodeIgniterTest.php:802]: Why not `ob_get_level()`?

Comment (kenjis) [reply tests/system/CodeIgniterTest.php]: Because I copied from the tearDown().

Changed.

**Review by MGatner — APPROVED** [review]

**Pull request merged**